WeatherMonitoringSystem.Tests/DomainTests/CoreTests/WeatherStationTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Moq;
+using FluentAssertions;
+using AutoFixture;
+using WeatherMonitoringSystem.Core;
+using WeatherMonitoringSystem.Bots;
+
+public class WeatherStationTests
+{
+    private readonly Fixture _fixture = new();
+    private readonly WeatherStation _weatherStation;
+
+    public WeatherStationTests()
+    {
+        _weatherStation = new WeatherStation();
+    }
+
+    [Fact]
+    public void AddBot_Should_Add_Bot_To_List()
+    {
+        var botMock = new Mock<IWeatherBot>();
+
+        _weatherStation.AddBot(botMock.Object);
+
+        _weatherStation.GetAllBots().Should().ContainSingle().Which.Should().Be(botMock.Object);
+    }
+
+    [Fact]
+    public void AddBot_Should_Throw_Exception_When_Bot_Is_Null()
+    {
+        Action act = () => _weatherStation.AddBot(null);
+
+        act.Should().Throw<ArgumentNullException>();
+    }
+
+
+    [Fact]
+    public void RemoveBot_Should_Remove_Bot_From_List()
+    {
+        var botMock = new Mock<IWeatherBot>();
+        _weatherStation.AddBot(botMock.Object);
+
+        _weatherStation.RemoveBot(botMock.Object);
+
+        _weatherStation.GetAllBots().Should().BeEmpty();
+    }
+
+    [Fact]
+    public void RemoveBot_Should_Throw_Exception_When_Bot_Is_Null()
+    {
+        Action act = () => _weatherStation.RemoveBot(null);
+
+        act.Should().Throw<ArgumentNullException>();
+    }
+
+    [Fact]
+    public void RemoveBot_Should_Not_Alter_List_When_Bot_Is_Not_In_List()
+    {
+        var botMock = new Mock<IWeatherBot>();
+
+        _weatherStation.RemoveBot(botMock.Object);
+
+        _weatherStation.GetAllBots().Should().BeEmpty();
```

Comment:
to test this case the right way, you should already have bots added and check the count before and after is the same